### ISSUE TITLE
Make web code public and fully automate deployment

### ISSUE THREAD

**Issue by sumpfork** [sumpfork/dominiontabs#102]
This is a bit of a note to myself and a bigger project, but I'd like to see
- the django project that runs the website be public in another git repo
- the deployment process of the web service be fully automated, i.e.
  - make the web site on my serving host
  - checkout out the relevant two repos
  - configure the website properly

Another option would be to move the service to something like AWS and use lambdas for the actual generation. I have quite a bit of experience with this and it might be cheaper and easier to deploy in the long run, but it would involve moving the whole existing website to AWS to be worthwhile.

**Comment by sumpfork** [comment]
There's a first attempt at a fully automatically deployed generator version online at http://bgtools.sandflea.org. Web and deployment code is [here](http://github.com/sumpfork/bgtools_web).

I simplified the expansion/edition options a little for the web generator so that you can't produce exceptions about 'no cards' as easily as with the command line script. I'm also not sure I've exposed everything properly that got added.

@wvoigt, @nickv2002 could you do some testing? Thanks.

**Comment by wvoigt** [comment]
I'm still trying to get my head around the code, so I didn't want to just make changes.  But from inspection, I found:

- On the actual [web page](http://bgtools.sandflea.org/) for some reason, I can only see the choices for **Expansion Selection**.  All the others are not expanded and therefore changes can't be made.
- `groupsets`, in effect, selects `--order` but only gives the choices "expansion" and "global".  But --order now has the following choices: ["expansion", "global", "colour", "cost"]
- `pagesize`, did you want to add A3?  I thought there was a request on that.

Missing options:

- `--types` (binary) has been added to print the card type at the top of the divider text (on the same line as the counts if that is selected.)  It makes sense to add this near the `counts` option.
- `--centre_expansion_dividers`, (binary) to Centre the tabs on expansion dividers. This is a sub option of `expansion_dividers`.
- `--expansion_dividers_long_name`, (binary) to use the long name with edition information on the                         expansion divider tab. Without this, the shorter expansion name is used on the expansion divider tab. This is a sub option of `expansion_dividers`.
- `--upgrade_with_expansion` (binary) to include any new edition upgrade cards with the                        expansion being upgraded.  This makes sense to be in the **Groups and Extras** section.
- `--base_cards_with_expansion` (binary) to print the base cards as part of the expansion (i.e., a                         divider for 'Silver' will be printed as both a 'Dominion' card and as an 'Intrigue 1st Edition'                         card). If this option is not given, all base cards are placed in their own 'Base' expansion. This makes sense to be in the **Groups and Extras** section.
- `--thickness` for wrappers only.  Thickness in cm of 60 'Copper' Cards.  Choices should be [2.0, 2.4, 3.2] default 2.0 if unsleeved.  default should be 2.4 if sleeved.  Alternatively, could use `--sleeved_thick` (binary) to represent the thick sleeves (aka 3.2) and `--sleeved_thin` (binary) to represent thinner sleeves (aka 2.4).  By default --thickness is 2.0 which fits unsleeved cards.  
Or better yet, maybe it makes sense to have the card size options as ['Sleeved-thin', 'Sleeved-thick', 'Unsleeved'] and then `options.size` and `options.sleeved_thick`, and` options.sleeved_thin` are all set based upon the `cardsize `selection..
- `--notch` (binary) for wrappers only.  Add a thumb notch to make it easier to remove the cards.

**Comment by sumpfork** [comment]
Oh, yeah, I somehow broke the accordion widget. Was still working earlier yesterday... not much to test then, sorry. Will try to find time to fix today.

**Comment by sumpfork** [comment]
Ok, fixed, I simply hadn't updated release mode to actually include the necessary javascript - I patched it for now, but will need to work on that.

**Comment by wvoigt** [comment]
Also, `events` seems to be defined, but not included anywhere for selection.  It should probably also go into **Groups and Extras**

**Comment by nickv2002** [comment]
On vacation with limited internet now. Will check next week. Great progress! 👍

**Comment by tttppp** [comment]
Maybe I missed a setting, but I couldn't see how to make a set of base dividers (i.e. just a set containing Copper, Silver, Gold, Platinum, Estate, etc.)

This is quite easy to do with the older (currently deployed) version.

Thanks for the great program btw!

**Comment by wvoigt** [comment]
@tttppp, yes that was oversight in the actual dominiontabs code.  I've updated EXPANSION_CHOICES to include 'base' now.  You won't see it on the new web interface until the dominiontabs code is brought over again.

**Comment by sumpfork** [comment]
I added `order` instead of `groupsets` as well as A3. Will add more options later, but just trying to get the thing out for now.

**Comment by nickv2002** [comment]
I poked around with the generator today and it seems pretty good. Some small things I noticed:

1. If I select 2 sets with Events (for example Promos and Empires) I get separate Promo dividers for each. Personally I put all my Events together since there's only 1 card in Promos and 1 for each event (rather than the usual 10).  I didn't see any web option to only make 1 Event divider. Not a big deal but a small personal preference.
1. Could the **Expansions to Include** section be a bit longer as it's getting to be a long list with lots of scrolling? Also might be good to remind people they can command/control click (depending on platform) to make noncontiguous selections there.  Finally, could it use the proper capitalized names instead of the short stub names?
1.  The icon for 'Hosted on ? Webfaction.' is missing

All of these are small things and so I think you're basically ready to launch this...

**Comment by wvoigt** [comment]
@nickv2002 when the `events` is brought out (aka "Exclude Individual Events & Landmarks"), then all events  from across all selected expansions will be grouped together.

**Comment by sumpfork** [comment]
Ok, I addressed your 2. and 3., @nickv2002, and everything's now live at http://domtabs.sandflea.org. Future deployment should be a lot more straightforward and consistent, as you staging/testing deployment. To note, I also added an automatically inserted version/git hash to the Changelog tag so we know what's actually live. Thanks again to everyone! Time to close some GitHub issues (and make new ones as folks discover new bugs :).
